WordPress/Classes/System/RootViewPresenter.swift

```diff
@@ -38,13 +38,35 @@ protocol RootViewPresenter: AnyObject {
 
     // MARK: Notifications
 
-    var notificationsViewController: NotificationsViewController? { get }
-    func showNotificationsTab()
-    func showNotificationsTabForNote(withID notificationID: String)
-    func switchNotificationsTabToNotificationSettings()
-    func popNotificationsTabToRoot()
+    func showNotificationsTab(completion: ((NotificationsViewController) -> Void)?)
 
     // MARK: Me
 
     func showMeScreen(completion: ((MeViewController) -> Void)?)
 }
+
+extension RootViewPresenter {
+
+    func showNotificationsTab() {
+        showNotificationsTab(completion: nil)
+    }
+
+    func showNotificationsTabForNote(withID notificationID: String) {
+        showNotificationsTab {
+            $0.showDetailsForNotificationWithID(notificationID)
+        }
+    }
+
+    func popNotificationsTabToRoot() {
+        showNotificationsTab {
+            $0.navigationController?.popToRootViewController(animated: false)
+        }
+    }
+
+    func switchNotificationsTabToNotificationSettings() {
```

Comment:
(nit) It should be fine to show it modally without switching to the notifications tab.

Reply:
I'm going to leave this for future consideration, as I'd like to make this part of the sidebar design fit into the app while maintaining the existing features of other parts of the app. 🙂